plugins/catch_pokemon/__init__.py

```diff
@@ -4,137 +4,144 @@
 import random
 import time
 
+from app import kernel
 from pokemongo_bot.human_behaviour import sleep
-from pokemongo_bot.event_manager import manager
-from pokemongo_bot import logger
 
 
-@manager.on("pokemon_found")
-def pokemon_found(bot, encounters=None):
-    # type: (PokemonGoBot, Optional[List[Encounter]]) -> None
+@kernel.container.register('catch_pokemon', ['@event_manager', '@logger'], tags=['plugin'])
+class CatchPokemon(object):
+    def __init__(self, event_manager, logger):
+        self.event_manager = event_manager
+        self.logger = logger
```

Comment:
Missed opportunity here: if most/all plugins will use logger, have plugins subclass from a Plugin class that properly wraps logger with appropriate prefix.